File: JEJE/프로그래머스-스킬트리.py
# https://programmers.co.kr/learn/courses/30/lessons/49993
def solution(skill, skill_trees):
    answer = 0
    dic = {}
    alph = ["Q","W","E","R","T","Y","U",'I','O','P','A','S','D','F','G','H','J','K','L','Z','X','C','V','B','N','M']
    for i in alph:
        dic[i] = 0
    for i, word in enumerate(list(skill), start = 1):
        dic[word] = i
    for i in skill_trees:
        count = 1
        for j in list(i):
            if dic[j] > count:
                count = 0
                break
            elif dic[j] == count:
                count +=1
            elif count == len(skill):
                break
        if count > 0: answer+=1
    return answer

File: JEJE/프로그래머스-스킬트리.cpp
// https://programmers.co.kr/learn/courses/30/lessons/49993?language=cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0, count = 0;
    map<char, int> m;
    for (int i = 0; i < skill.size(); i++)
    {
        m[skill[i]] = i + 1;
    }
    for (auto &i : skill_trees) {
        int cnt = 1;
        for (char &j : i)
        {
            int soon = m[j];
            if (soon == cnt) cnt += 1;
            else if (soon > cnt) { cnt = 0; break; }
            else if (cnt == skill.size() && 1 != skill.size()) break;
        }
        if (cnt > 0) answer += 1;
    }
    return answer;
}